gym_solo/core/rewards.py

```diff
@@ -102,6 +141,48 @@ def compute(self) -> float:
       float: A real-valued number in [-1, 1], where 1 means perfectly upright 
       whilst -1 means that the robot is literally upside down. 
     """
-    _, quat = p.getBasePositionAndOrientation(self._robot_id)
-    unused_x, y, unused_z = np.array(p.getEulerFromQuaternion(quat))
+    _, quat = self.client.getBasePositionAndOrientation(self._robot_id)
+    unused_x, y, unused_z = np.array(self.client.getEulerFromQuaternion(quat))
     return self._fully_upright * y / self._fully_upright ** 2
+
+
+class HomePositionReward(Reward):
+  """Rewards the robot for being in the home position. Currently, this rewards
+  the robot for being orientated properly and being as tall as possible. This
+  will require some experimentation--if the robot ends up jumping to maximize
+  the reward, this might need to be modified to a more intelligent height 
+  reward.
+  """
+  # Found via experimentation--should be close enough to prevent any 
+  # non-trivial errors
+  _quad_standing_height = 0.3
+  _max_angle = np.pi
+
+  def __init__(self, robot_id: int):
+    """Create a new HomePositionReward.
+
+    Args:
+      robot_id (int): The PyBullet body id of the robot
+    """
+    self._robot_id = robot_id
+
+  def compute(self) -> float:
+    """Compute the HomePositionReward for the current state. 
+
+    Returns:
+      float: A real-valued in [0, 1], where 1 is in the home position.
+    """
+    (unused_x, unused_y, z), quat = self.client.getBasePositionAndOrientation(
+      self._robot_id)
+    theta_x, theta_y, unused_z = np.array(
```

Comment:
Why are we using quat if we get x,y,z in the previous statement

Reply:
that x,y,z is for position, `quat` is for orientation

Reply:
In the next line you derive x,y,z from quat right? My question was as to why are you calculating that in the next line if you can extract it here itself

Reply:
so `getBasePositionAndOrientation` returns `(pos, orien)`. `pos` is a tuple of `(x, y, z)` and `orien` is in quaternions. Since we need both the position as well as the orientation of the bot, (and cause I don't know how to do quaternion math lol), I extract the euler angles `(Θ_x, Θ_y, Θ_z)`.

The `(x, y, z)` conversion isn't really happening twice as one's for position while the other is for orientation.

Reply:
Oh me dum dum!! My brain did not work